verification/cbmc/proofs/aws_cryptosdk_priv_hdr_parse_aad/aws_cryptosdk_priv_hdr_parse_aad_harness.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cryptosdk/header.h>
+#include <aws/cryptosdk/private/header.h>
+#include <make_common_data_structures.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+#include <proof_helpers/utils.h>
+
+void make_hash_table_with_no_backing_store(struct aws_hash_table *map, size_t max_table_entries);
+
+static int flag = 0;
+
+/**
+ * In the aws_cryptosdk_enc_ctx_deserilize() proof, the first value we read (second overall) is the number of elements,
+ * which we need to be constrained in order to ensure that the proof finishes. All other values can be left nondet.
+ * This generates exactly that set of bytes
+ */
+uint16_t aws_byte_cursor_read_be16_generator_for_parse_aads(const struct aws_byte_cursor *cursor) {
+    (void)cursor;
```

Comment:
I am curious: This is to avoid warnings regarding non-usage of cursor here, right?

Reply:
That's what I thought as well. This stub is taken and modified from the `aws_enc_ctx_deserialize` proof so that's where the original idea came from.